DSLCompiler: Add the project directory to improve static analysis

SurenNihalani opened the pull request:
Why: helps with LinkedIn's internal migration

What changed: added a projectDirectory which helps us identify the subproject in a MP

Tests performed: modified unit tests

ericsun2 reviewed (commented):
projectDirectory might be a little bit fuzzy here, since "project" is also used in Azkaban deployment.

Maybe "repositoryDirectory" or "repositoryPath" is slightly better?

ericsun2 commented on hadoop-plugin/src/test/groovy/com/linkedin/gradle/azkaban/AzkabanDslYamlCompilerTest.groovy:
```diff
@@ -79,6 +79,9 @@ class AzkabanDslYamlCompilerTest {
 
     Project mockProject = mock(Project.class);
     when(mockProject.name).thenReturn("test");
+    when(mockProject.getRootDir()).thenReturn(new File("/home/root_directory"))
```
This path works on Linux but not on Mac.
/var/tmp/root_directory may work on both

SurenNihalani replied:
It's not meant to be a real path. won't be resolved

SurenNihalani committed:
Azkaban projects: include project directory to help with migration estimation

Why: helps with static analysis

What changed: added logic to export relative subproject directory in
buildMetadata.json

Design choices (if any):  preferring to put it here instead of flows
because it's consistent across a project

Tests performed: integration tests were modified to accomodate this
change
